src/universe/me.ts

```diff
@@ -0,0 +1,172 @@
+import { Client, ClientError } from '../client'
+import { StaffMessageTemplateFavorite } from '../entities/staff/staff'
+import { throwExceptionFromCommonError } from '../helpers'
+import { MeData, Universe, UniverseBadRequestError, UniverseForbiddenError, UniverseMeError, UniverseMePreferencesError, UniverseSessionError } from '.'
+import { PatchOpts } from 'src/helpers/json-patch-type'
+
+type addTemplateFavoriteArg = Omit<StaffMessageTemplateFavorite, 'created_at'>
+type removeTemplateFavoriteArg = Omit<StaffMessageTemplateFavorite, 'created_at'> & {created_at?: string}
+export interface UniverseMeProps {
+  fetch: (skipInterceptors?: boolean) => Promise<MeData | never>
+  session: () => Promise<MeData | never>
+  addTemplateFavorite: (favorite: addTemplateFavoriteArg) => Promise<MeData | never>
+  removeTemplateFavorite: (favorite: removeTemplateFavoriteArg) => Promise<MeData | never>
+}
+
+export class UniverseMe implements UniverseMeProps {
+  private cachedMeData?: MeData
+  private readonly universe: Universe
+  private readonly http: Client
+  constructor ({ universe, http }: {universe: Universe, http: Client}) {
+    this.universe = universe
+    this.http = http
+  }
+
+  private async patchStaffPreferences (opts: PatchOpts): Promise<MeData | never> {
+    try {
+      const response = await this.http.getClient()(opts)
+
+      const updatedStaff = response?.data?.data?.[0]
+
+      if (!this.cachedMeData || !this.cachedMeData.user) throw new UniverseMeError()
+
+      this.setCachedMeData({
+        ...this.cachedMeData,
+        preferences: updatedStaff.preferences,
+        staff: updatedStaff
+      })
+
+      return this.cachedMeData
+    } catch (error) {
+      throwExceptionFromCommonError(error as ClientError)
+      throw new UniverseMePreferencesError(undefined, { error })
+    }
+  }
+
+  private async getPatchConfig (): Promise<{
+    staffRequestOpts: PatchOpts | null
+    meData: MeData
+  }> {
+    const me = this.cachedMeData ?? await this.fetch()
+
+    return {
+      staffRequestOpts: me.staff.id ? {
+        method: 'PATCH',
+        url: `${this.universe.universeBase}/api/v0/staff/${me.staff.id}`,
+        data: []
+      } : null,
+      meData: me
+    }
+  }
+
+  private setCachedMeData (data?: MeData | null): Universe {
```

Comment:
since this is private now, we can safely update it to our needs. We don't need to return Unversse here, let's return `void`